win32ss/user/ntuser/nonclient.c

```diff
@@ -400,56 +416,20 @@ DefWndDoSizeMove(PWND pwnd, WORD wParam)
       }
       else if (g_bWindowSnapEnabled && msg.message == WM_LBUTTONUP)
       { // If WindowSnapEnabled: Decide whether to snap before exiting
-         DWORD ExStyleTB, StyleTB;
-         BOOL IsTaskBar;
-
-         // We want to forbid snapping operations on the TaskBar
-         // We use a heuristic for detecting the TaskBar Wnd by its typical Style & ExStyle Values
-         ExStyleTB = (ExStyle & WS_EX_TOOLWINDOW);
-         StyleTB = (Style & (WS_POPUP | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN));
-         IsTaskBar = (StyleTB == (WS_POPUP | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN))
-                     && (ExStyleTB == WS_EX_TOOLWINDOW);
-         TRACE("ExStyle=%x Style=%x IsTaskBar=%d\n", ExStyleTB, StyleTB, IsTaskBar);
-
-         // check for snapping if was moved by caption
-         if (!IsTaskBar && hittest == HTCAPTION && thickframe && (ExStyle & WS_EX_MDICHILD) == 0)
+         if (hittest == HTCAPTION && thickframe && /* Check for snapping if was moved by caption */
+             IntIsSnapAllowedForWindow(pwnd) && (ExStyle & WS_EX_MDICHILD) == 0)
          {
-            RECT snapRect;
-            BOOL doSideSnap = FALSE;
-            UserSystemParametersInfo(SPI_GETWORKAREA, 0, &snapRect, 0);
-
-            // snap to left
-            if (pt.x <= snapRect.left)
-            {
-               snapRect.right = (snapRect.right - snapRect.left) / 2 + snapRect.left;
-               doSideSnap = TRUE;
-            }
-            // snap to right
-            if (pt.x >= snapRect.right-1)
-            {
-               snapRect.left = (snapRect.right - snapRect.left) / 2 + snapRect.left;
-               doSideSnap = TRUE;
-            }
-
-            if (doSideSnap)
-            {
-               co_WinPosSetWindowPos(pwnd,
-                                     NULL,
-                                     snapRect.left,
-                                     snapRect.top,
-                                     snapRect.right - snapRect.left,
-                                     snapRect.bottom - snapRect.top,
-                                     SWP_NOACTIVATE);
-               pwnd->InternalPos.NormalRect = origRect;
-            }
-            else
+            BOOLEAN wasSnap = IntIsWindowSnapped(pwnd); /* Need the live snap state, not orgSnap nor maximized state */
```

Comment:
No, it means live as in, is the HWND snapped right now. Meaning in this case, if it was snapped to the left and the user drags the window so it snaps to the right, it should not set NormalRect to the old position because we don't want to store a snap position there.

Reply:
ah yes, "live" in that sense, sorry :joy: